src/IRrecv.h

```diff
@@ -51,8 +51,16 @@ const uint16_t kMaxTimeoutMs = kRawTick * (UINT16_MAX / MS_TO_USEC(1));
 const uint32_t kFnvPrime32 = 16777619UL;
 const uint32_t kFnvBasis32 = 2166136261UL;
 
+#if DECODE_AC
 // Hitachi AC is the current largest state size.
 const uint16_t kStateSizeMax = kHitachiAc2StateLength;
+#elif DECODE_MWM
```

Comment:
Possibly best/simplest to just add `DECODE_MWM` to the `#define DECODE_AC` mega OR/|| line.

Reply:
My concern here was to avoid allocating 53 bytes here for anybody who only needs the 18 of MWM. I added some code to handle multiple lengths, so if you'd rather keep the 53, I can remove that code, too.

Reply:
I _used_ to have a `MAX()` #define for calculating the maximum size. That eventually exploded due to size. If you've got a nice elegant size function, I'm all for it. Maybe in a different PR?

I used to worry about the 53 bytes, but then I stopped. If someone is that desperate they can change it from 53 ;-)

Reply:
It wasn't elegant at all, just an `#elif DECODE_MWM`.